component/grade-table.js

```diff
@@ -10,48 +11,68 @@ class GradeTable {
+            }            
+    
+        }
+    }
 
-            var $tr = document.createElement("tr");
-            
-            var gradesItem = grades[i];
-
-            var $tdName = document.createElement("td");
-            var $tdNameText = document.createTextNode(gradesItem.name);
-            $tdName.appendChild($tdNameText);
-
-            var $tdCourse = document.createElement("td");
-            var $tdCourseText = document.createTextNode(gradesItem.course);
-            $tdCourse.appendChild($tdCourseText);
 
-            var $tdGrade = document.createElement("td");
-            var $tdGradeText = document.createTextNode(gradesItem.grade);
-            $tdGrade.appendChild($tdGradeText);
+        onDeleteClick (deleteGrade){
+            this.deleteGrade = deleteGrade;
+        }
 
-            $tr.append($tdName, $tdCourse, $tdGrade);
 
-            this.tableElement.appendChild($tr);
+        renderGradeRow(data, deleteGrade) {
+
+            for (var i=0; i<data.length; i++) {
+    
+                var $tr = document.createElement("tr");
+    
+                //var gradesItem = grades[i];
+                var gradesItem = data[i];
+    
+                var $tdName = document.createElement("td");
+                var $tdNameText = document.createTextNode(gradesItem.name);
+                $tdName.appendChild($tdNameText);
+
+                var $tdCourse = document.createElement("td");
+                var $tdCourseText = document.createTextNode(gradesItem.course);
+                $tdCourse.appendChild($tdCourseText);
+
+                var $tdGrade = document.createElement("td");
+                var $tdGradeText = document.createTextNode(gradesItem.grade);
+                $tdGrade.appendChild($tdGradeText);
+    
+                // $tr.append($tdName, $tdCourse, $tdGrade);
+
+                var $tdDelete = document.createElement("td");
+                var $tdDeleteButton = document.createElement("button");
+                $tdDeleteButton.textContent = "Delete";
+                $tdDeleteButton.className = "btn btn-primary";
+                $tdDeleteButton.addEventListener('click', function() { deleteGrade(gradesItem.id);
+                });
+    
+                
+    
+                $tdDelete.appendChild($tdDeleteButton);
+                $tr.append($tdName, $tdCourse, $tdGrade, $tdDelete);
+                this.tableElement.appendChild($tr);
+
+                           
             
 
-
-            // var $tdName = $("<td>", {
-            //     class: "tdName",
-            //     text: grades.name[i]
-            // })
-
-            // var $tCourse = $("<td>", {
-            //     class: "tdCourse",
-            //     text: grades.course[i]
-            // })
-
-            // var $tdGrade = $("<td>", {
-            //     class: "tdGrade",
-            //     text: grades.grade[i]
-            // })           
-
         }
-
+                    
     }

```

Comment:
As stated above, this method should create a single table row (and its contents) and return it, not appending it to the table body.